### ISSUE TITLE
C error when filling slice with return value

### ISSUE THREAD

**Issue by bstnbuck** [vlang/v#10505]
**V version:** V 0.2.2 de384f1
**OS:** all (tested on Windows and Linux)

**What did you do?**
```v
fn main(){
	mut test := []int{len: 10}
	test[0..3] << tf()
	println(test)
}

fn tf() []int{
	return [1,2,3]
}
```
Compiling with TCC works, but it is not filled properly

**What did you expect to see?**
`[1,2,3,0,0,0,0,0,0,0]` or V error
 
**What did you see instead?**
Compiling with [gcc, msvc, mingw]:
C error like this:
```
==================
/tmp/v/test.9984698556038889805.tmp.c: In function ‘main__main’:
/tmp/v/test.9984698556038889805.tmp.c:10102:13: error: lvalue required as unary ‘&’ operand
10102 |  _PUSH_MANY(&array_slice(test, 0, 2), (main__tf()), _t731, Array_int);
      |             ^
/tmp/v/test.9984698556038889805.tmp.c:124:82: note: in definition of macro ‘_PUSH_MANY’
  124 | #define _PUSH_MANY(arr, val, tmp, tmp_typ) {tmp_typ tmp = (val); array_push_many(arr, tmp.data, tmp.len);}
      |                                                                                  ^~~
...
```

**Comment by UweKrueger** [comment]
I can reproduce the compilation error with `gcc` and `clang` &ndash; compilation works with `tcc` for me, too. I've found the reason and I'm working on a fix. As a workaround you can write:
```v
fn main(){
    mut test := []int{len: 10}
    mut a := test[0..3]
    a << tf()
    println(test)
    println(a)
}

fn tf() []int{
    return [1,2,3]
}
```
This works with all compilers and prints the result:
`[0, 0, 0, 0, 0, 0, 0, 0, 0, 0]`
`[0, 0, 0, 1, 2, 3]`

The only difference to your example is that the slice is given a name `a` &ndash; so the result of appending to it becomes accessible. This result is correct and as expected. The behaviour has been implemented in #10219 and is documented [here](https://github.com/vlang/v/blob/master/doc/docs.md#array-slices).

In brief it says that slicing creates an alternative view to a part of an existing array that becomes independent (i.e. cloned) in the moment when new elements are added (I've called this _"copy on grow")_. This is intentionally different from Go's behaviour.

**Comment by billpcs** [comment]
:rocket: It seems that there is no crash anymore. 

#10219 has been merged, so it makes sense.
